src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,492 @@
+
+        return { objects: objects, pager: pager };
+    }
+
+    private async getATCItems(namespace: string) {
+        return (await this.globalStorageClient.getObject<ATCItem[]>(namespace)) ?? [];
+    }
+
+    async getUserModules(config: Config): Promise<GLASSModule[]> {
+        const modules = await this.getModules();
+
+        return getUserModules(modules, config.currentUser);
+    }
+
+    async delete(namespace: string, items: Id[]): Promise<void> {
+        await this.deleteFileResources(items);
+
+        const uploads = await this.getUploads(namespace);
+        const updatedUploads = this.updateStatusById(items, uploads, "DELETED");
+        return await this.globalStorageClient.saveObject<GLASSDataMaintenanceItem[]>(namespace, updatedUploads);
+    }
+
+    async getColumns(namespace: string): Promise<string[]> {
+        const columns = await this.storageClient.getObject<string[]>(namespace);
+
+        return columns ?? [];
+    }
+
+    async saveColumns(namespace: string, columns: string[]): Promise<void> {
+        return this.storageClient.saveObject<string[]>(namespace, columns);
+    }
+
+    async uploadATC(namespace: string, file: File, year: string, items: ATCItemIdentifier[]): Promise<void> {
+        const atcItems = await this.getATCItems(namespace);
+        const jsons = await this.extractJsonFromZIP(file);
+
+        if (jsons.length === 4) {
+            if (items) {
+                const updatedVersion = this.updateVersion(items);
+                const updatedATCItems = this.patchATCVersion(atcItems, items);
+
+                await this.globalStorageClient.saveObject<ATCJson>(`ATC-${updatedVersion}`, jsons);
+                await this.globalStorageClient.saveObject<ATCJson>(Namespaces.ATCS, updatedATCItems);
+            } else {
+                const updatedATCItems = this.uploadNewATC(year, atcItems);
+
+                await this.globalStorageClient.saveObject<ATCJson>(`ATC-${year}-v1`, jsons);
+                await this.globalStorageClient.saveObject<ATCJson>(Namespaces.ATCS, updatedATCItems);
+            }
+        } else {
+            throw new Error("The zip file does not contain exactly 4 JSON files.");
+        }
+    }
+
+    async getRecalculationLogic(namespace: string): Promise<AMCRecalculation | undefined> {
+        return await this.globalStorageClient.getObject<AMCRecalculation>(namespace);
+    }
+
+    async cancelRecalculation(namespace: string): Promise<void> {
+        const amcRecalculationLogic = await this.getRecalculationLogic(namespace);
+
+        if (amcRecalculationLogic) {
+            const updatedRecalculationLogic = {
+                ...amcRecalculationLogic,
+                recalculate: false,
+            };
+
+            await this.globalStorageClient.saveObject<AMCRecalculation>(namespace, updatedRecalculationLogic);
+        }
+    }
+
+    async saveRecalculationLogic(namespace: string, atcNamespace: string): Promise<void> {
```

Comment:
This function does not fulfill the solid principle of single responsibility: the name of the function makes me think that only amc-recalculations is saved in the data store, but you are also updating ATCs. It's better to split here in the repository in two different functions and both functions are called from the use case that is in charge of coordinating them. Also rename the use case according to what it has to do